### PR TITLE
feat: split contact name into first and last name across the application

### PR DESCRIPTION
**Pull request by YasinduRC**
- Updated JSON language files to reflect changes in contact name fields.
- Modified components to handle firstName and lastName instead of a single name field.
- Adjusted validation rules for first and last names.
- Implemented utility functions to retrieve full contact names.
- Updated tests to accommodate changes in contact structure.

## Summary
- Splits the CRM `Contact` name field into `firstName` (required) and `lastName` (optional) across the frontend.
- Updates Add/Edit Contact modals with side-by-side First name / Last name inputs, inline validation on blur.
- Updates all contact display surfaces (contact table, side panels, deal/task cards, kanban board, dropdown options) to render the combined full name via a shared `getContactFullName` helper.
- Updates types, constants, regex, and validation schemas to match the new field split.

> **Note:** Backend API changes and the corresponding database migration for this feature were implemented and merged in a separate PR. This PR contains frontend changes only.

### How to test

- [ ] Add Contact: first name required, last name optional, inline validation on blur
- [ ] Edit Contact: existing contacts load first/last name correctly, partial updates save only changed fields
- [ ] Contact table shows full name (first + last), sorts correctly
- [ ] Deal/task side panels and kanban cards show correct contact name
- [ ] Existing unit tests pass (`contactUtil`, `dealUtil`, `kanbanUtil`)

### Project Checklist

- [ ] Changes build without any errors
- [ ] Have written adequate test cases
- [ ] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [ ] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [ ] Pull request is raised from the correct source branch
- [ ] Pull request is raised to the correct destination branch
- [ ] Pull request is raised with correct title
- [ ] Pull request is self reviewed
- [ ] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR cleanly splits contact name into firstName/lastName across types, forms, validations, and all display sites; type changes are consistent and no stale CrmContact.name references remain. The main concern is a form-validation UX regression in ContactModalForm: switching to validateOnBlur while removing the touched-gating causes errors on all required fields to surface as soon as any single field is blurred. Two minor items: missing unit tests for the new getContactFullName helper and an inconsistency in how empty lastName is submitted between create and edit.

Found **6** new issue(s): 🔴 1 important, 🟡 4 suggestion(s), 🟣 1 nit(s)

File: frontend/src/community/crm/components/molecules/ContactModalForm/ContactModalForm.tsx [hunk 1]
```diff
@@ -168,19 +168,36 @@ const ContactModalForm = ({
...
-        maxLength={CONTACT_NAME_MAX_LENGTH}
-        required
-        fullWidth
-      />
+      <div className="flex flex-row gap-[0.625rem]">
+        <InputField
+          name="firstName"
+          value={values.firstName}
+          errorMessage={errors.firstName}
+          state={errors.firstName ? "error" : "default"}
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/ContactModalForm/ContactModalForm.tsx:176]: 🤖 **Claude** · 🔴 **Important**

Regression in error display timing. The form switched from validateOnChange:true/validateOnBlur:false with touched-gated errors to validateOnBlur:true, but all `touched.<field>` checks were removed so errors now render purely from `errors.<field>`. Formik runs a full-schema validation on every blur and populates the entire `errors` object, so blurring just the first name field will immediately show 'Email is required', 'Owner is required', etc. on fields the user has not yet interacted with. Re-introduce touched-gating while keeping onBlur wired, e.g. `state={touched.firstName && errors.firstName ? 'error' : 'default'}` and `errorMessage={touched.firstName ? errors.firstName : undefined}` (and likewise for lastName, email, contactNumber, ownerId). Destructure `touched` again from formik.

File: frontend/src/community/crm/utils/contactUtil.ts [hunk 2]
```diff
@@ -7,6 +7,14 @@ import {
 } from "~community/crm/types/CommonTypes";
 import { groupItemsByPriority } from "~community/crm/utils/crmUtil";
 
+interface ContactNameParts {
+  firstName?: string | null;
+  lastName?: string | null;
+}
+
+export const getContactFullName = (contact?: ContactNameParts | null): string =>
```
Comment (ThinuwanW) [inline frontend/src/community/crm/utils/contactUtil.ts:15]: 🤖 **Claude** · 🟡 **Suggestion**

The new `getContactFullName` helper is now used in ~12 places and has non-trivial edge behavior (null/undefined contact, null lastName, empty firstName, trimming/joining), but has no unit test. Add tests in contactUtil.test.ts covering: firstName + lastName -> 'John Smith', firstName + null lastName -> 'John', empty firstName + lastName -> 'Smith', and undefined/null contact -> ''.

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx [hunk 3]
```diff
@@ -78,7 +79,8 @@ const AddContactModalContent = () => {
 
   const createContact = (values: CrmContactFormValues) => {
     const payload: CrmContactCreatePayload = {
-      name: values.name.trim(),
+      firstName: values.firstName.trim(),
+      lastName: values.lastName.trim() || undefined,
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: 🤖 **Claude** · 🟣 **Nit**

Inconsistent empty-lastName handling between create and edit. Create sends `values.lastName.trim() || undefined` (field omitted when empty), while EditContactModalContent sends the trimmed empty string `""` via getChangedContactFields when a previously-set lastName is cleared. Confirm the backend treats `""` and an omitted/undefined lastName the same way for clearing; if not, normalize the edit path to send `undefined`/`null` for a cleared lastName so both paths behave consistently.

File: frontend/src/community/crm/constants/contactConstants.ts [hunk 4]
```diff
@@ -2,6 +2,6 @@ export const CONTACT_SEARCH_DEBOUNCE_DELAY = 500;
 export const DEFAULT_PAGE_SIZE = 10;
 export const DEFAULT_COMPANY_PAGE_SIZE = 100;
 export const ALL_COMPANIES = undefined;
-export const CONTACT_NAME_MAX_LENGTH = 255;
+export const CONTACT_NAME_MAX_LENGTH = 50;
```
Comment (ThinuwanW) [inline frontend/src/community/crm/constants/contactConstants.ts:5]: 🤖 **Claude** · 🟡 **Suggestion**

CONTACT_NAME_MAX_LENGTH was reduced 255 -> 50 and is now applied via the shared `addContactValidations` schema to BOTH first and last name AND to the Edit form (ContactModalForm is reused for edit). Any existing contact whose migrated name segment exceeds 50 characters will fail Yup validation and can no longer be saved through Edit, even if the user never touches the name field (validation runs on the whole form). Confirm the backend enforces the same 50-char limit and that legacy data was migrated/truncated; if not, consider only validating fields the user actually changed, or relaxing the limit for edits.

File: frontend/src/community/crm/regex/crmRegexPatterns.ts [hunk 5]
```diff
@@ -7,5 +7,5 @@ export function isValidCompanyWebsiteUrl(): RegExp {
 }
 
 export function isContactNameValid(): RegExp {
-  return /^[\p{L} \-.,']+$/u;
+  return /^[\p{L} \-.']+$/u;
```
Comment (ThinuwanW) [inline frontend/src/community/crm/regex/crmRegexPatterns.ts]: 🤖 **Claude** · 🟡 **Suggestion**

The allowed-characters pattern for contact names dropped the comma (`/^[\p{L} \-.,']+$/u` -> `/^[\p{L} \-.']+$/u`). Because this schema is shared by the Edit form, any pre-existing contact whose first/last name contains a comma (previously valid, e.g. "Smith, Jr") will now trigger the invalid-characters error and become unsaveable via Edit even without editing the name. If commas were intentionally disallowed, ensure legacy names were migrated; otherwise re-allow the comma or validate only changed fields.

File: frontend/src/community/crm/utils/contactUtil.ts [hunk 6]
```diff
@@ -7,6 +7,14 @@ import {
 } from "~community/crm/types/CommonTypes";
 import { groupItemsByPriority } from "~community/crm/utils/crmUtil";
 
+interface ContactNameParts {
```
Comment (ThinuwanW) [inline frontend/src/community/crm/utils/contactUtil.ts:10]: 🤖 **Claude** · 🟡 **Suggestion**

`getContactFullName` is a new helper consumed by ~12 components with non-trivial edge behavior (null/undefined contact, null lastName, empty firstName, whitespace trimming via join+trim), yet no unit test was added — the sibling test files (dealUtil/kanbanUtil/contactUtil) were only touched to update fixtures. Add explicit cases: null/undefined contact -> "", firstName only -> no trailing space, lastName only -> no leading space, both present -> single-space join, to lock in the display contract that many UI paths now depend on.

File: frontend/src/community/crm/components/molecules/ContactModalForm/ContactModalForm.tsx (hunk 1 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/molecules/ContactModalForm/ContactModalForm.tsx:176]: This was intentional to remove touch and validateOnFocus:true

File: frontend/src/community/crm/utils/contactUtil.ts (hunk 2 above)
Comment (YasinduRC) [reply frontend/src/community/crm/utils/contactUtil.ts:15]: Added unit tests

File: frontend/src/community/crm/utils/contactUtil.ts (hunk 6 above)
Comment (YasinduRC) [reply frontend/src/community/crm/utils/contactUtil.ts:10]: Added unit tests

File: frontend/src/community/crm/regex/crmRegexPatterns.ts (hunk 5 above)
Comment (YasinduRC) [reply frontend/src/community/crm/regex/crmRegexPatterns.ts]: Will  check in the migration endpoint is we have contact names with commas in the prod, will update with results of that check

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx (hunk 3 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: Fixed in the backend - Backend now check the string is empty via stringUtils and set null if it is empty

File: frontend/src/community/crm/utils/contactUtil.ts (hunk 6 above)
Comment (anusham-rc) [inline frontend/src/community/crm/utils/contactUtil.ts:10]: can replace this with concatStrings util

File: frontend/src/community/crm/components/molecules/DeleteContactModalContent/DeleteContactModalContent.tsx [hunk 7]
```diff
@@ -6,6 +6,7 @@ import { useToast } from "~community/common/providers/ToastProvider";
 import { useDeleteContact } from "~community/crm/api/ContactApi";
 import CrmDeleteModalContent from "~community/crm/components/molecules/CrmDeleteModalContent/CrmDeleteModalContent";
 import { useCrmStore } from "~community/crm/store/store";
+import { getContactFullName } from "~community/crm/utils/contactUtil";
```
Comment (anusham-rc) [inline frontend/src/community/crm/components/molecules/DeleteContactModalContent/DeleteContactModalContent.tsx:9]: can replace this with concatStrings

File: frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx [hunk 8]
```diff
@@ -60,23 +60,26 @@ const EditContactModalContent = () => {
   );
 
   const initialValues: CrmContactFormValues = {
-    name: selectedContact?.name,
-    email: selectedContact?.email,
+    firstName: selectedContact?.firstName ?? "",
+    lastName: selectedContact?.lastName ?? "",
+    email: selectedContact?.email ?? "",
```
Comment (anusham-rc) [inline frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx:65]: check this why optional

File: frontend/src/community/crm/utils/contactValidations.ts [hunk 9]
```diff
@@ -14,14 +14,29 @@ type TranslatorFunctionType = (suffixes: string[]) => string;
...
+        CONTACT_NAME_MAX_LENGTH,
+        translator(["validations", "firstNameLength"])
+      )
       .matches(isContactNameValid(), {
-        message: translator(["validations", "nameInvalidCharacters"]),
+        message: translator(["validations", "firstNameInvalidCharacters"]),
         excludeEmptyString: true
       })
-      .required(translator(["validations", "name"])),
+      .required(translator(["validations", "firstName"])),
```
Comment (anusham-rc) [inline frontend/src/community/crm/utils/contactValidations.ts]: check the validation order, required should be first

File: frontend/src/community/crm/utils/dealUtil.ts [hunk 10]
```diff
@@ -29,11 +30,14 @@ export const buildOwnerOptions = <T extends OptionSource>(
 export const buildContactOptions = (
   items: CrmContactLookup[]
 ): DropdownOption[] => {
-  const toOption = (item: CrmContactLookup): DropdownOption => ({
-    id: item.id,
-    value: item.id,
-    label: item.company?.name ? `${item.name} ${item.company.name}` : item.name
-  });
+  const toOption = (item: CrmContactLookup): DropdownOption => {
+    const fullName = getContactFullName(item);
```
Comment (anusham-rc) [inline frontend/src/community/crm/utils/dealUtil.ts]: check this

**Review by anusham-rc — CHANGES_REQUESTED** [review]

File: frontend/src/community/crm/utils/contactValidations.ts (hunk 9 above)
Comment (YasinduRC) [reply frontend/src/community/crm/utils/contactValidations.ts]: Moved to the top

File: frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx (hunk 8 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx:65]: the type is string so it will be string | undefined, this was not handled in the previous implementation

File: frontend/src/community/crm/components/molecules/DeleteContactModalContent/DeleteContactModalContent.tsx (hunk 7 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/molecules/DeleteContactModalContent/DeleteContactModalContent.tsx:9]: cannot replace this with concatStrings since it does not have  it has no handling for an empty part, so when there's no last name it emits a trailing space , which getContactFullName avoids. They are not equivalent.

File: frontend/src/community/crm/utils/contactUtil.ts (hunk 6 above)
Comment (YasinduRC) [reply frontend/src/community/crm/utils/contactUtil.ts:10]: cannot replace this with concatStrings since it does not have  it has no handling for an empty part, so when there's no last name it emits a trailing space , which getContactFullName avoids. They are not equivalent.

File: frontend/src/community/crm/utils/dealUtil.ts (hunk 10 above)
Comment (YasinduRC) [reply frontend/src/community/crm/utils/dealUtil.ts]: used the fullName inline

File: frontend/src/community/crm/constants/contactConstants.ts (hunk 4 above)
Comment (YasinduRC) [reply frontend/src/community/crm/constants/contactConstants.ts:5]: We need to discuss this with the team. I checked with Nethma and asked them to see whether there's a way to identify any names in the production environment that violate this rule. If that's not possible, we should update the character limit to 100.

File: frontend/src/community/crm/utils/contactUtil.ts [hunk 11]
```diff
@@ -7,6 +7,14 @@ import {
...
+  firstName?: string | null;
```
Comment (thusala) [inline frontend/src/community/crm/utils/contactUtil.ts]: ideally firstName can't be null at any point

**Review by thusala — CHANGES_REQUESTED** [review]

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx (hunk 3 above)
Comment (thusala) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: ideally i think u should pass "" instead of undefined for this scenario. could u verify that?

File: frontend/src/community/crm/utils/contactUtil.ts (hunk 11 above)
Comment (YasinduRC) [reply frontend/src/community/crm/utils/contactUtil.ts]: Fixed, the rootcause was the deal reponse dto sended the contactID,firstName and lastName seperatley instead of a contactlookup. Updated the backend as well

File: frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx (hunk 3 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/molecules/AddContactModalContent/AddContactModalContent.tsx]: Now the inconsistency between create and edit for the optional lastName field is fixed now. Both paths now use null to represent "no last name", while omission in PATCH means "unchanged". Added JsonNullable to CrmContact's lastName in the edit request so the backend can distinguish between an omitted field and an explicit clear (null).

**Review by thusala — DISMISSED** [review]

File: frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx [hunk 12]
```diff
@@ -60,23 +60,26 @@ const EditContactModalContent = () => {
...
       email: values.email.trim(),
       contactNumber: values.contactNumber.trim(),
       companyId: values.companyId,
       ownerId: values.ownerId
     };
     const originalValues: CrmContactFormValues = {
-      name: selectedContact?.name.trim() ?? "",
+      firstName: selectedContact?.firstName.trim() ?? "",
+      lastName: selectedContact?.lastName?.trim() ?? "",
       email: selectedContact?.email.trim() ?? "",
```
Comment (akilarootcode) [inline frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx]: add optional chaining for firstName and email both

File: frontend/src/community/crm/components/molecules/TaskModalForm/TaskModalForm.tsx [hunk 13]
```diff
@@ -194,7 +193,7 @@ const TaskModalForm: FC<TaskFormProps> = ({
       (contactLookupItem) => String(contactLookupItem.id) === item.id
     );
     formik.setFieldValue("contactId", Number(item.id));
-    setSelectedContactName(contact?.name ?? String(item.content));
+    setSelectedContactName(getContactFullName(contact) || String(item.content));
```
Comment (akilarootcode) [inline frontend/src/community/crm/components/molecules/TaskModalForm/TaskModalForm.tsx]: remove this fallback String(item.content)

File: frontend/src/community/crm/components/organisms/DealSidePanel/DealPropertiesSidebar.tsx [hunk 14]
```diff
@@ -101,7 +102,7 @@ const DealPropertiesSidebar: FC<DealPropertiesSidebarProps> = ({
   };
 
   const handleContactChange = (contact: CrmContactLookup | null): void => {
-    if (contact && contact.id !== deal.contactId) {
+    if (contact && contact.id !== deal.contact?.id) {
```
Comment (akilarootcode) [inline frontend/src/community/crm/components/organisms/DealSidePanel/DealPropertiesSidebar.tsx:105]: remove this and make contact.id optional

File: frontend/src/community/crm/types/CommonTypes.ts [hunk 15]
```diff
@@ -89,7 +89,8 @@ export interface EditCompanyPayload extends CrmCompanyCreatePayload {
 
 export interface CrmContactType {
   id: number;
-  name: string;
+  firstName: string;
+  lastName: string | null;
```
Comment (akilarootcode) [inline frontend/src/community/crm/types/CommonTypes.ts:93]: remove null and make optional

File: frontend/src/community/crm/types/CommonTypes.ts [hunk 16]
```diff
@@ -136,15 +138,17 @@ export interface CrmCompanyDomainSearchResponseType {
...
   email: string;
   contactNumber: string;
   companyId: number | null;
   ownerId: number | null;
 }
 
 export interface CrmContactCreatePayload {
-  name: string;
+  firstName: string;
+  lastName: string | null;
```
Comment (akilarootcode) [inline frontend/src/community/crm/types/CommonTypes.ts:151]: remove null

File: frontend/src/community/crm/types/CommonTypes.ts [hunk 17]
```diff
@@ -153,7 +157,8 @@ export interface CrmContactCreatePayload {
 
 export interface EditContactPayload {
   id?: number;
-  name?: string;
+  firstName?: string;
+  lastName?: string | null;
```
Comment (akilarootcode) [inline frontend/src/community/crm/types/CommonTypes.ts:161]: remove null

File: frontend/src/community/crm/utils/contactUtil.ts [hunk 18]
```diff
@@ -7,6 +7,14 @@ import {
 } from "~community/crm/types/CommonTypes";
 import { groupItemsByPriority } from "~community/crm/utils/crmUtil";
 
+interface ContactNameParts {
+  firstName: string;
+  lastName?: string | null;
```
Comment (akilarootcode) [inline frontend/src/community/crm/utils/contactUtil.ts:12]: remove null if we have optional

**Review by akilarootcode — CHANGES_REQUESTED** [review]

File: frontend/src/community/crm/components/molecules/TaskModalForm/TaskModalForm.tsx (hunk 13 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/molecules/TaskModalForm/TaskModalForm.tsx]: Valid, and it's already fixed in a other PR

File: frontend/src/community/crm/types/CommonTypes.ts (hunk 16 above)
Comment (YasinduRC) [reply frontend/src/community/crm/types/CommonTypes.ts:151]: cannot remove the null since it creates the payload for the API

File: frontend/src/community/crm/types/CommonTypes.ts (hunk 17 above)
Comment (YasinduRC) [reply frontend/src/community/crm/types/CommonTypes.ts:161]: cannot remove the null since it creates the payload for the API

File: frontend/src/community/crm/components/organisms/DealSidePanel/DealPropertiesSidebar.tsx (hunk 14 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/organisms/DealSidePanel/DealPropertiesSidebar.tsx:105]: They're not equivalent. contact && ... skips the comparison when contact is null/undefined, whereas contact?.id !== deal.contact?.id still compares undefined to deal.contact?.id, which can evaluate to true and change the behavior.

File: frontend/src/community/crm/utils/contactUtil.ts (hunk 18 above)
Comment (YasinduRC) [reply frontend/src/community/crm/utils/contactUtil.ts:12]: lastName?: string | null is intentional here because this is a shared input type. Real DTOs always have a lastName key whose value may be null, while some callers (including tests) omit the property entirely. Using both ? and | null allows the interface to accept both shapes.

File: frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx (hunk 12 above)
Comment (YasinduRC) [reply frontend/src/community/crm/components/molecules/EditContactModalContent/EditContactModalContent.tsx]: Added the optional chaining to them

File: frontend/src/community/crm/types/CommonTypes.ts (hunk 15 above)
Comment (YasinduRC) [reply frontend/src/community/crm/types/CommonTypes.ts:93]: lastName is intentionally typed as string | null because "no last name" is a valid business state in this application. The validation schema allows it, and the backend returns null for that case. Changing the type to string would incorrectly tell TypeScript the value is always present, hiding legitimate null checks and making it easier to introduce runtime errors. The type reflects the actual API contract, not just the happy path.

**Review by akilarootcode — APPROVED** [review]

**Review by anusham-rc — APPROVED** [review]
